api/src/main/java/org/apache/iceberg/expressions/BoundAggregate.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.expressions;
+
+import org.apache.iceberg.StructLike;
+
+public class BoundAggregate<T> extends Aggregate<BoundTerm<T>> implements Bound<T> {
```

Comment:
I don't think these types are correct. If the aggregate is over a `BoundTerm` that produces `T`, there is no guarantee that the aggregate also produces `T`. For example, `Count` may wrap `BoundTerm<String>` and produce an `Integer`.